# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1288769159 14400
#      Wed Nov 03 03:25:59 2010 -0400
# Node ID a512c537b38e1289e05e94c6e471a86c152056d8
# Parent  1a26199cb21227db8354b6b7cb1dc3ec4c217c51
cquad.cc: define copysign if needed

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2010-11-03  John W. Eaton  <jwe@octave.org>
+
+	* bootstrap.conf (gnulib_modules): Include copysign in the list.
+
 2010-10-28  Rik  <octave@nomad.inbox5.com>
 
 	* NEWS: Update deprecated function list with autocov and autocor.
 
 2010-10-25  Kai Habel  <kai.habel@gmx.de>
 
 	* NEWS: Add uimenu.
 
diff --git a/bootstrap.conf b/bootstrap.conf
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -14,16 +14,17 @@
 
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 
 # gnulib modules used by this package.
 gnulib_modules="
   c-strcase
+  copysign
   crypto/md5
   fcntl
   fnmatch
   getcwd
   gethostname
   getopt-gnu
   gettimeofday
   glob
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-11-03  John W. Eaton  <jwe@octave.org>
+
+	* DLD-FUNCTIONS/cquad.cc: If we don't have copysign but we do
+	have _copysign, define copysign to be _copysign.
+
 2010-11-02  John W. Eaton  <jwe@octave.org>
 
 	* DLD-FUNCTIONS/cquad.cc (Fcquad): Use octave_Inf, not INFINITY.
 
 2010-11-02  David Bateman  <dbateman@free.fr>
 
 	* ls-hdf5.cc (bool hdf5_get_scalar_attr (hid_t, hid_t,
         const char *, void *), herr_t hdf5_add_scalar_attr
diff --git a/src/DLD-FUNCTIONS/cquad.cc b/src/DLD-FUNCTIONS/cquad.cc
--- a/src/DLD-FUNCTIONS/cquad.cc
+++ b/src/DLD-FUNCTIONS/cquad.cc
@@ -26,16 +26,21 @@ along with Octave; see the file COPYING.
 
 #include <stdlib.h>
 #include "lo-math.h"
 #include "lo-ieee.h"
 #include "oct.h"
 #include "parse.h"
 #include "ov-fcn-handle.h"
 
+#if ! defined (HAVE_COPYSIGN) && defined (HAVE__COPYSIGN)
+#define copysign _copysign
+#define HAVE_COPYSIGN 1
+#endif
+
 /* Define the size of the interval heap. */
 #define cquad_heapsize                  200
 
 
 /* Data of a single interval */
 typedef struct
 {
   double a, b;
